src/MainWindow.vala

```diff
@@ -535,9 +524,25 @@ namespace Terminal {
             add (grid);
 
             menu_popover.closed.connect (() => {
+                var binding = menu_popover.get_data<Binding> ("zoom-binding");
+                binding.unref ();
                 current_terminal.grab_focus ();
             });
 
+            menu_button.pressed.connect (() => {
+                zoom_default_button.label = font_scale_to_zoom (current_terminal.font_scale);
+                var binding = current_terminal.bind_property ("font-scale",
+                                                              zoom_default_button,
+                                                              "label",
+                                                              BindingFlags.DEFAULT,
+                                                              (binding, from_val, ref to_val) => {
+                                                to_val.set_string (font_scale_to_zoom (from_val.get_double ()));
+                                                return true;
+                                                              });
+
+                menu_popover.set_data<Binding> ("zoom-binding", binding);
+            });
```

Comment:
I think something like this would more closely follow our whitespace conventions and is much easier to read:

```suggestion
            menu_button.pressed.connect (() => {
                zoom_default_button.label = font_scale_to_zoom (current_terminal.font_scale);
                var binding = current_terminal.bind_property (
                    "font-scale",
                    zoom_default_button,
                    "label",
                    BindingFlags.DEFAULT,

                    (binding, from_val, ref to_val) => {
                        to_val.set_string (font_scale_to_zoom (from_val.get_double ()));
                        return true;
                    }
                );

                menu_popover.set_data<Binding> ("zoom-binding", binding);
            });
```